test-utils/src/lib.rs

```diff
@@ -0,0 +1,152 @@
+use portpicker::pick_unused_port;
+use rand::{distributions::Alphanumeric, Rng};
+use std::{
+    process::Command,
+    thread::sleep,
+    time::{Duration, SystemTime},
+};
+use uuid::Uuid;
+
+/// A docker instance for a postgres database. It should be used
+/// by tests as a singleton (e.g. once_cell::sync::Lazy), and any
+/// test logic that connects to it should separate cases by creating
+/// an unique database. Also, the instance should implement a destructor.
+///
+/// Example usage:
+///
+/// static PG: Lazy<PostgresDockerInstance> = Lazy::new(PostgresDockerInstance::default);
+
+/// #[dtor]
+/// fn cleanup() {
+///    PG.cleanup();
+/// }
+///
+///  #[tokio::test]
+/// async fn test_case() {
+///     // Create a unique database name so we have a new database for each test.
+///     let db_name = Uuid::new_v4().to_string();
+///     let db_uri = PG.get_unique_uri(db_name.as_str());
+///     
+///     // Test logic below, which can use `db_uri` to connect to the postgres instance.
+/// }
+///
```

Comment:
This will give code highlighting.

```suggestion
/// ```
/// static PG: Lazy<PostgresDockerInstance> = Lazy::new(PostgresDockerInstance::default);

/// #[dtor]
/// fn cleanup() {
///    PG.cleanup();
/// }
///
///  #[tokio::test]
/// async fn test_case() {
///     // Create a unique database name so we have a new database for each test.
///     let db_name = Uuid::new_v4().to_string();
///     let db_uri = PG.get_unique_uri(db_name.as_str());
///     
///     // Test logic below, which can use `db_uri` to connect to the postgres instance.
/// }
/// ```
```

Reply:
Done.